pkg/controller/installation/installation_controller.go

```diff
@@ -202,17 +211,44 @@ func (r *ReconcileInstallation) Reconcile(request reconcile.Request) (reconcile.
 		return reconcile.Result{}, err
 	}
 
-	retryRequeue := reconcile.Result{
-		Requeue:      true,
-		RequeueAfter: 10 * time.Second,
-	}
-
 	//context is cancelled on delete of an installation, so if context is cancelled but there is no deletion timestamp
 	//the installation must be created after one was deleted, so recreate a context to use for the new installation.
 	if r.context.Err() == context.Canceled && installation.DeletionTimestamp == nil {
 		r.context, r.cancel = context.WithCancel(context.Background())
 	}
 
+	rhmiSubscription, err := r.getIntegreatlyOperatorSubscription(request.NamespacedName.Namespace)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	if r.isRHMIUpgradeAvailable(rhmiSubscription) {
+		//TODO expose event since RHMI operator will be redeployed and logs will likely get lost
+		//TODO Setup secondary watch on subscription resources in the rhmi operator namespace
+		logrus.Infof("RHMI upgrade available")
+
+		latestRHMIInstallPlan, err := r.getIntegreatlyOperatorInstallPlan(rhmiSubscription)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+
+		latestRHMICSV, err := r.getIntegreatlyOperatorCSV(latestRHMIInstallPlan)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+
+		if !r.isRHMIUpgradeServiceAffecting(latestRHMICSV) {
+			err = r.approveRHMIUpgrade(latestRHMIInstallPlan)
```

Comment:
A case where a previous upgrade is still in progress is not covered here.

What if we have a non service affecting release out, before scheduled upgrade of a service affecting release happened? 
Once the scheduled time comes, we will approve first InstallPlan, then second InstallPlan(for non affecting release) will be created shortly after first one is applied. The second installPlan might be applied before operator finished all of it's intended upgrade actions(for first upgrade).

Also, ignoring the possibility of two upgrades in a row, we should consider checking that RHMI status.phase is PhaseCompleted, as it's probably not a good idea to roll out upgrade if previous reconcile didn't finish completely.

Reply:
Good spot @matskiv . I will check the phaseCompleted here also. 

In the case of handling multiple upgrades in a row. this will be covered in this spike JIRA - https://issues.redhat.com/browse/INTLY-7273

Reply:
@davidffrench I was concerned that, when a second upgrade is available, operator might be restarted before it finishes first upgrade, because it will approve non service affecting InstallPlan at the beginning of reconcile loop, and new deployment will be rolled out, killing the operator mid-reconcile.
I don't think INTLY-7273 addressed this specific concern.

Reply:
Based on our research today, OLM will only ever create one pending approval install plan for a subscription and moving the logic to the after the products reconcilers should ensure a second upgrade does not take place until the first is complete.